google-cloud-firestore/lib/google/cloud/firestore/document/reference.rb

```diff
@@ -0,0 +1,418 @@
+          #
+          #   # Create a document
+          #   nyc_ref.create({ name: "New York City" })
+          #
+          def create data
+            ensure_context!
+
+            if context.respond_to? :create
+              context.create self, data
+            else
+              context.database.create self, data
+            end
+          end
+
+          ##
+          # Write to document with the provided object values. If the document
+          # does not exist, it will be created. By default, the provided data
+          # overwrites existing data, but the provided data can be merged into
+          # the existing document using the `merge` argument.
+          #
+          # @param [Hash] data The document's fields and values.
+          # @param [true, String|Symbol, Array<String|Symbol>] merge When
+          #   provided and `true` all data is merged with the existing document
+          #   data. When provided only the specified as a list of field paths
+          #   are merged with the existing document data. The default is to
+          #   overwrite the existing document data.
+          #
+          # @return [Time] The time the changes were committed
+          #
+          # @example
+          #   require "google/cloud/firestore"
+          #
+          #   firestore = Google::Cloud::Firestore.new
+          #
+          #   # Get a document reference
+          #   nyc_ref = firestore.doc "cities/NYC"
+          #
+          #   # Set a document
+          #   nyc_ref.set({ name: "New York City" })
+          #
+          # @example Set a document and merge all data:
+          #   require "google/cloud/firestore"
+          #
+          #   firestore = Google::Cloud::Firestore.new
+          #
+          #   # Get a document reference
+          #   nyc_ref = firestore.doc "cities/NYC"
+          #
+          #   # Set a document
+          #   nyc_ref.set({ name: "New York City" }, merge: true)
+          #
+          # @example Set a document and merge only name:
+          #   require "google/cloud/firestore"
+          #
+          #   firestore = Google::Cloud::Firestore.new
+          #
+          #   # Get a document reference
+          #   nyc_ref = firestore.doc "cities/NYC"
+          #
+          #   # Set a document
+          #   nyc_ref.set({ name: "New York City" }, merge: [:name])
+          #
+          def set data, merge: nil
+            ensure_context!
+
+            if context.respond_to? :set
+              context.set self, data, merge: merge
+            else
+              context.database.set self, data, merge: merge
+            end
+          end
+
+          ##
+          # Write to document with the provided object values. If the document
+          # does not exist, it will be created. By default, the provided data
+          # overwrites existing data, but the provided data can be merged into
+          # the existing document using the `merge` argument.
```

Comment:
There doesn't seem to be a merge argument or example. I thought the service did this automatically for updates?

Reply:
The `#update` method always merges, and there is no argument to control the behavior. 

The `#set` method replaces by default, but can merge when the `merge` argument is provided. It does have additional code examples demonstrating how to merge.

Reply:
Oops, I see what you are pointing out now. You are correct, this part of the `#update` method's documentation was accidentally duplicated from the `#set` method's documentation. This has been fixed.